File: backend-old/src/core/face_data.py
from schema.face_schema import Face, FaceType, Ring, Spot

face_data: list[Face] = [
    Face(
        face_type=FaceType.NONE,
        face_name="No Target Face",
        render_cross=False,
        viewBox=0.0,
        spots=[],
        rings=[],
    ),
    Face(
        face_name="WA 122cm Standard Target Face",
        face_type=FaceType.WA_122_FULL,
        render_cross=True,
        viewBox=1342.0,
        spots=[
            Spot(
                x_offset=0.0,
                y_offset=0.0,
                diameter=1220.0,
            )
        ],
        rings=[
            Ring(
                data_score=1,
                fill="#FFFFFF",
                r=610.0,  # 1220mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=2,
                fill="#FFFFFF",
                r=549.0,  # 1098mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=3,
                fill="#000000",
                r=488.0,  # 976mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=4,
                fill="#000000",
                r=427.0,  # 854mm / 2
                stroke="#FFFFFF",
                stroke_width=1.0,
            ),
            Ring(
                data_score=5,
                fill="#00B4E4",
                r=366.0,  # 732mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=6,
                fill="#00B4E4",
                r=305.0,  # 610mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=7,
                fill="#F65058",
                r=244.0,  # 488mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=8,
                fill="#F65058",
                r=183.0,  # 366mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=9,
                fill="#FFE552",
                r=122.0,  # 244mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=61.0,  # 122mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=30.5,  # 61mm diameter / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
        ],
    ),
    Face(
        face_name="WA 80cm Standard Target Face",
        face_type=FaceType.WA_80_FULL,
        render_cross=True,
        viewBox=880.0,
        spots=[
            Spot(
                x_offset=0.0,
                y_offset=0.0,
                diameter=800.0,
            )
        ],
        rings=[
            Ring(
                data_score=1,
                fill="#FFFFFF",
                r=400.0,  # 800mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=2,
                fill="#FFFFFF",
                r=360.0,  # 720mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=3,
                fill="#000000",
                r=320.0,  # 640mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=4,
                fill="#000000",
                r=280.0,  # 560mm / 2
                stroke="#FFFFFF",
                stroke_width=1.0,
            ),
            Ring(
                data_score=5,
                fill="#00B4E4",
                r=240.0,  # 480mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=6,
                fill="#00B4E4",
                r=200.0,  # 400mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=7,
                fill="#F65058",
                r=160.0,  # 320mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=8,
                fill="#F65058",
                r=120.0,  # 240mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=9,
                fill="#FFE552",
                r=80.0,  # 160mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=40.0,  # 80mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=20.0,  # 40mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
        ],
    ),
    Face(
        face_name="WA 60cm Standard Target Face",
        face_type=FaceType.WA_60_FULL,
        render_cross=True,
        viewBox=660.0,
        spots=[
            Spot(
                x_offset=0.0,
                y_offset=0.0,
                diameter=600.0,
            )
        ],
        rings=[
            Ring(
                data_score=1,
                fill="#FFFFFF",
                r=300.0,  # 600mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=2,
                fill="#FFFFFF",
                r=270.0,  # 540mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=3,
                fill="#000000",
                r=240.0,  # 480mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=4,
                fill="#000000",
                r=210.0,  # 420mm / 2
                stroke="#FFFFFF",
                stroke_width=1.0,
            ),
            Ring(
                data_score=5,
                fill="#00B4E4",
                r=180.0,  # 360mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=6,
                fill="#00B4E4",
                r=150.0,  # 300mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=7,
                fill="#F65058",
                r=120.0,  # 240mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=8,
                fill="#F65058",
                r=90.0,  # 180mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=9,
                fill="#FFE552",
                r=60.0,  # 120mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=30.0,  # 60mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=15.0,  # 30mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
        ],
    ),
    Face(
        face_name="WA 40cm Standard Target Face",
        face_type=FaceType.WA_40_FULL,
        render_cross=True,
        viewBox=440.0,
        spots=[
            Spot(
                x_offset=0.0,
                y_offset=0.0,
                diameter=400.0,
            )
        ],
        rings=[
            Ring(
                data_score=1,
                fill="#FFFFFF",
                r=200.0,  # 400mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=2,
                fill="#FFFFFF",
                r=180.0,  # 360mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=3,
                fill="#000000",
                r=160.0,  # 320mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=4,
                fill="#000000",
                r=140.0,  # 280mm / 2
                stroke="#FFFFFF",
                stroke_width=1.0,
            ),
            Ring(
                data_score=5,
                fill="#00B4E4",
                r=120.0,  # 240mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=6,
                fill="#00B4E4",
                r=100.0,  # 200mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=7,
                fill="#F65058",
                r=80.0,  # 160mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=8,
                fill="#F65058",
                r=60.0,  # 120mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=9,
                fill="#FFE552",
                r=40.0,  # 80mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=20.0,  # 40mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=10.0,  # 20mm / 2 (recurve X)
                stroke="#000000",
                stroke_width=1.0,
            ),
        ],
    ),
    Face(
        face_name="WA 122cm 6-Ring Target Face",
        face_type=FaceType.WA_122_6RINGS,
        render_cross=True,
        viewBox=854.0,
        spots=[
            Spot(
                x_offset=0.0,
                y_offset=0.0,
                diameter=732.0,  # 5-ring outer diameter (366mm radius * 2)
            )
        ],
        rings=[
            Ring(
                data_score=5,
                fill="#00B4E4",
                r=366.0,  # 732mm / 2 (outermost for 6-ring)
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=6,
                fill="#00B4E4",
                r=305.0,  # 610mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=7,
                fill="#F65058",
                r=244.0,  # 488mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=8,
                fill="#F65058",
                r=183.0,  # 366mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=9,
                fill="#FFE552",
                r=122.0,  # 244mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=61.0,  # 122mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=30.5,  # 61mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
        ],
    ),
    Face(
        face_name="WA 80cm 6-Ring Target Face",
        face_type=FaceType.WA_80_6RINGS,
        render_cross=True,
        viewBox=560.0,
        spots=[
            Spot(
                x_offset=0.0,
                y_offset=0.0,
                diameter=480.0,  # 5-ring outer diameter (240mm radius * 2)
            )
        ],
        rings=[
            Ring(
                data_score=5,
                fill="#00B4E4",
                r=240.0,  # 480mm / 2 (outermost for 6-ring)
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=6,
                fill="#00B4E4",
                r=200.0,  # 400mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=7,
                fill="#F65058",
                r=160.0,  # 320mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=8,
                fill="#F65058",
                r=120.0,  # 240mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=9,
                fill="#FFE552",
                r=80.0,  # 160mm / 2
                stroke="#000000",
                stroke_width=2.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=40.0,  # 80mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
            Ring(
                data_score=10,
                fill="#FFE552",
                r=20.0,  # 40mm / 2
                stroke="#000000",
                stroke_width=1.0,
            ),
        ],
    ),
]

File: backend/internal/repository/face_data.go
package repository

import "github.com/jpmolinamatute/arch-stats/backend/internal/model"

// DefaultFaceCatalog contains the standardized World Archery target face definitions
// ported from face_data.py (14KB Python source).
var DefaultFaceCatalog = []model.FaceRead{
	{
		FaceType:    model.FaceTypeNone,
		FaceName:    "No Target Face",
		RenderCross: false,
		ViewBox:     0.0,
		Spots:       []model.Spot{},
		Rings:       []model.Ring{},
	},
	{
		FaceType:    model.FaceTypeWA122Full,
		FaceName:    "WA 122cm Standard Target Face",
		RenderCross: true,
		ViewBox:     1342.0,
		Spots: []model.Spot{
			{
				XOffset:  0.0,
				YOffset:  0.0,
				Diameter: 1220.0,
			},
		},
		Rings: []model.Ring{
			{DataScore: 1, Fill: "#FFFFFF", R: 610.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 2, Fill: "#FFFFFF", R: 549.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 3, Fill: "#000000", R: 488.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 4, Fill: "#000000", R: 427.0, Stroke: "#FFFFFF", StrokeWidth: 1.0},
			{DataScore: 5, Fill: "#00B4E4", R: 366.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 6, Fill: "#00B4E4", R: 305.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 7, Fill: "#F65058", R: 244.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 8, Fill: "#F65058", R: 183.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 9, Fill: "#FFE552", R: 122.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 10, Fill: "#FFE552", R: 61.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 10, Fill: "#FFE552", R: 30.5, Stroke: "#000000", StrokeWidth: 1.0},
		},
	},
	{
		FaceType:    model.FaceTypeWA80Full,
		FaceName:    "WA 80cm Standard Target Face",
		RenderCross: true,
		ViewBox:     880.0,
		Spots: []model.Spot{
			{
				XOffset:  0.0,
				YOffset:  0.0,
				Diameter: 800.0,
			},
		},
		Rings: []model.Ring{
			{DataScore: 1, Fill: "#FFFFFF", R: 400.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 2, Fill: "#FFFFFF", R: 360.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 3, Fill: "#000000", R: 320.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 4, Fill: "#000000", R: 280.0, Stroke: "#FFFFFF", StrokeWidth: 1.0},
			{DataScore: 5, Fill: "#00B4E4", R: 240.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 6, Fill: "#00B4E4", R: 200.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 7, Fill: "#F65058", R: 160.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 8, Fill: "#F65058", R: 120.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 9, Fill: "#FFE552", R: 80.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 10, Fill: "#FFE552", R: 40.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 10, Fill: "#FFE552", R: 20.0, Stroke: "#000000", StrokeWidth: 1.0},
		},
	},
	{
		FaceType:    model.FaceTypeWA60Full,
		FaceName:    "WA 60cm Standard Target Face",
		RenderCross: true,
		ViewBox:     660.0,
		Spots: []model.Spot{
			{
				XOffset:  0.0,
				YOffset:  0.0,
				Diameter: 600.0,
			},
		},
		Rings: []model.Ring{
			{DataScore: 1, Fill: "#FFFFFF", R: 300.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 2, Fill: "#FFFFFF", R: 270.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 3, Fill: "#000000", R: 240.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 4, Fill: "#000000", R: 210.0, Stroke: "#FFFFFF", StrokeWidth: 1.0},
			{DataScore: 5, Fill: "#00B4E4", R: 180.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 6, Fill: "#00B4E4", R: 150.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 7, Fill: "#F65058", R: 120.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 8, Fill: "#F65058", R: 90.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 9, Fill: "#FFE552", R: 60.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 10, Fill: "#FFE552", R: 30.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 10, Fill: "#FFE552", R: 15.0, Stroke: "#000000", StrokeWidth: 1.0},
		},
	},
	{
		FaceType:    model.FaceTypeWA40Full,
		FaceName:    "WA 40cm Standard Target Face",
		RenderCross: true,
		ViewBox:     440.0,
		Spots: []model.Spot{
			{
				XOffset:  0.0,
				YOffset:  0.0,
				Diameter: 400.0,
			},
		},
		Rings: []model.Ring{
			{DataScore: 1, Fill: "#FFFFFF", R: 200.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 2, Fill: "#FFFFFF", R: 180.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 3, Fill: "#000000", R: 160.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 4, Fill: "#000000", R: 140.0, Stroke: "#FFFFFF", StrokeWidth: 1.0},
			{DataScore: 5, Fill: "#00B4E4", R: 120.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 6, Fill: "#00B4E4", R: 100.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 7, Fill: "#F65058", R: 80.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 8, Fill: "#F65058", R: 60.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 9, Fill: "#FFE552", R: 40.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 10, Fill: "#FFE552", R: 20.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 10, Fill: "#FFE552", R: 10.0, Stroke: "#000000", StrokeWidth: 1.0},
		},
	},
	{
		FaceType:    model.FaceTypeWA1226Rings,
		FaceName:    "WA 122cm 6-Ring Target Face",
		RenderCross: true,
		ViewBox:     854.0,
		Spots: []model.Spot{
			{
				XOffset:  0.0,
				YOffset:  0.0,
				Diameter: 732.0,
			},
		},
		Rings: []model.Ring{
			{DataScore: 5, Fill: "#00B4E4", R: 366.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 6, Fill: "#00B4E4", R: 305.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 7, Fill: "#F65058", R: 244.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 8, Fill: "#F65058", R: 183.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 9, Fill: "#FFE552", R: 122.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 10, Fill: "#FFE552", R: 61.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 10, Fill: "#FFE552", R: 30.5, Stroke: "#000000", StrokeWidth: 1.0},
		},
	},
	{
		FaceType:    model.FaceTypeWA806Rings,
		FaceName:    "WA 80cm 6-Ring Target Face",
		RenderCross: true,
		ViewBox:     560.0,
		Spots: []model.Spot{
			{
				XOffset:  0.0,
				YOffset:  0.0,
				Diameter: 480.0,
			},
		},
		Rings: []model.Ring{
			{DataScore: 5, Fill: "#00B4E4", R: 240.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 6, Fill: "#00B4E4", R: 200.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 7, Fill: "#F65058", R: 160.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 8, Fill: "#F65058", R: 120.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 9, Fill: "#FFE552", R: 80.0, Stroke: "#000000", StrokeWidth: 2.0},
			{DataScore: 10, Fill: "#FFE552", R: 40.0, Stroke: "#000000", StrokeWidth: 1.0},
			{DataScore: 10, Fill: "#FFE552", R: 20.0, Stroke: "#000000", StrokeWidth: 1.0},
		},
	},
}
